### PR TITLE
add more unit tests to matrix converter

### PR DESCRIPTION
**Pull request by parthshahva**

### CODE REVIEW

File: tests/unit/docker/test_matrix_converter.py [hunk 1]
```diff
@@ -93,6 +175,28 @@ def test__load_cell_table_slice(self, mock_open, mock_pd_read_csv):
...
 
+    @mock.patch("s3fs.S3FileSystem.open")
+    def test__load_expression_table_slice(self, mock_open):
+        manifest_file_path = "tests/functional/res/expression_metadata_manifest"
+        with open(manifest_file_path) as f:
+            mock_open.return_value = f
+            self.matrix_converter.expression_manifest = self.matrix_converter._parse_manifest("test_manifest_key")
+        results = self.matrix_converter._load_expression_table_slice(0)
+        mock_open.assert_called_once()
+        self.assertEqual(type(results).__name__, 'generator')
```
Comment (calvinnhieu) [inline tests/unit/docker/test_matrix_converter.py]: Does it make sense here to verify some values in the generator?

**Comment by calvinnhieu** [comment]
Given the refactoring done in `matrix_converter`, can you verify these changes pass functional tests on dev?

File: tests/unit/docker/test_matrix_converter.py (hunk 1 above)
Comment (parthshahva) [reply tests/unit/docker/test_matrix_converter.py]: hmm I think its fine the way it is

**Comment by parthshahva** [comment]
This has been tested on dev

**Pull request merged**